libraries/abstractions/pkcs11/utest/iot_pkcs11_mbedtls_utest.c

```diff
@@ -96,32 +169,90 @@ int suiteTearDown( int numFailures )
     return numFailures;
 }
 
-/* ======================  TESTING C_Initialize  ============================ */
+/* ==========================  Helper functions  ============================ */
 
 /*!
- * @brief C_Initialize happy path.
+ * @brief Helper function to initialize PKCS #11.
  *
  */
-void test_pkcs11_C_Initialize( void )
+static CK_RV prvInitializePkcs11()
 {
     CK_RV xResult = CKR_OK;
 
-    /* Hack: Giving out pointer to xResult when creating a mutex. This unit testcase
-     * will never use the actual mutex. */
-    xQueueCreateMutex_IgnoreAndReturn( ( SemaphoreHandle_t ) &xResult );
+    xQueueCreateMutex_IgnoreAndReturn( 1 );
     CRYPTO_Init_Ignore();
     mbedtls_entropy_init_Ignore();
     mbedtls_ctr_drbg_init_Ignore();
     mbedtls_ctr_drbg_seed_IgnoreAndReturn( 0 );
     xResult = C_Initialize( NULL );
 
-    TEST_ASSERT_EQUAL( CKR_OK, xResult );
+    return xResult;
+}
+
+/*!
+ * @brief Helper function to initialize PKCS #11.
+ *
+ */
+static CK_RV prvUninitializePkcs11()
+{
+    CK_RV xResult = CKR_OK;
 
     mbedtls_entropy_free_CMockIgnore();
     mbedtls_ctr_drbg_free_CMockIgnore();
     vQueueDelete_CMockIgnore();
     xResult = C_Finalize( NULL );
 
+    return xResult;
+}
+
+/*!
+ * @brief Helper function to create a PKCS #11 session.
+ *
+ */
+static CK_RV prvOpenSession( CK_SESSION_HANDLE_PTR pxSession )
+{
+    CK_RV xResult = CKR_OK;
+    CK_FLAGS xFlags = CKF_SERIAL_SESSION | CKF_RW_SESSION;
+
+    pvPortMalloc_Stub( pvPkcs11MallocCb );
+    xQueueCreateMutex_IgnoreAndReturn( ( SemaphoreHandle_t ) &xResult );
+    xQueueCreateMutex_IgnoreAndReturn( ( SemaphoreHandle_t ) &xResult );
+    xResult = C_OpenSession( 0, xFlags, NULL, 0, pxSession );
+
+    return xResult;
+}
+
+/*!
+ * @brief Helper function to close a PKCS #11 session.
+ *
+ */
+static CK_RV prvCloseSession( CK_SESSION_HANDLE_PTR pxSession )
+{
+    CK_RV xResult = CKR_OK;
+
+    vQueueDelete_Ignore();
+    vQueueDelete_Ignore();
+    vPortFree_Stub( vPkcs11FreeCb );
+    mbedtls_sha256_free_CMockIgnore();
+    xResult = C_CloseSession( *pxSession );
+
+    return xResult;
+}
+
+/* ======================  TESTING C_Initialize  ============================ */
+
+/*!
+ * @brief C_Initialize happy path.
+ *
+ */
+void test_pkcs11_C_Initialize( void )
+{
+    CK_RV xResult = CKR_OK;
+
+    xResult = prvInitializePkcs11();
```

Comment:
it would be much clearer what this function is trying to do if the contents of the initialize were right here instead of in a function, since this is the purpose of this function testing initialize

It is ok for the content of the deinitialize  to be in a function since it is only a helper including its assert

Reply:
Done